tests/test_pipenv.py

```diff
@@ -539,6 +539,43 @@ def test_sequential_mode(self, pypi):
             c = p.pipenv('run python -c "import requests; import idna; import certifi; import records; import tpfd; import parse;"')
             assert c.return_code == 0
 
+    @pytest.mark.install
+    @pytest.mark.run
+    def test_pep423_name_install(self, pypi):
+        with PipenvInstance(pypi=pypi) as p:
+            with open(p.pipfile_path, 'w') as f:
+                contents = """
+[packages]
+python_dateutil = "*"
```

Comment:
can you add a test case that has an unnormalized name and confirm that it stays unnormalized and comments are preserved when you install a new package that doesn't overlap?

Reply:
can easily check this by adding to a couple more lines to the pipfile

```
# Pre-comment - I should totally be preserved
requests = "*"  # Inline comment - should definitely be preserved too!
```